source/expressions.tex

```diff
@@ -3354,13 +3354,8 @@
 for class template deduction\iref{over.match.class.deduct}
 for the remainder of this subclause.
 Otherwise, if the type is \tcode{auto},
-it is replaced by the type deduced for the variable \tcode{x}
-in the invented declaration\iref{dcl.spec.auto},
-which is never interpreted as a function declaration:
-\begin{codeblock}
-auto x @\placeholder{init}@;
-\end{codeblock}
-, where \placeholder{init} is the initializer.
+it is replaced by the type
+determined by placeholder type deduction\iref{dcl.type.auto.deduct}.
```

Comment:
Even more radical alternative:

> Otherwise, if the type contains a placeholder type, it is replaced by the type
determined by placeholder type deduction\iref{dcl.type.auto.deduct}.

And then move the "shall" up to dcl.type.auto.deduct:

> For an explicit type conversion, `T` is the specified type, which shall be `auto`.